pandas/core/frame.py

```diff
@@ -4279,6 +4279,11 @@ def check_int_infer_dtype(dtypes):
                     # error: Argument 1 to "append" of "list" has incompatible type
                     # "Type[signedinteger[Any]]"; expected "Type[signedinteger[Any]]"
                     converted_dtypes.append(np.int64)  # type: ignore[arg-type]
+                elif dtype == "float":
+                    # GH#42452 : np.dtype("float") coerces to np.float64 from Numpy 1.20
+                    converted_dtypes.extend(
+                        [np.float64, np.float32, np.float16]  # type: ignore[list-item]
```

Comment:
I think here @jreback means we don't explicitly support `np.float64`. We would just like to capture np.float32/64